### PR TITLE
[Quantity value] Add endpoints for units

### LINKED ISSUE #579: [Quantity value] Add endpoints
Add pendant for the following endpoints from classic bundle:

- `/admin/quantity-value/unit-list`
- `/admin/quantity-value/convert`
- `/admin/quantity-value/convert-all`

part of https://github.com/pimcore/studio-backend-bundle/issues/566

### PR DESCRIPTION
**Pull request by lukmzig**
## Changes in this pull request
Resolves #579 

## Additional info
Added pendant for the following endpoints from classic bundle:

- /admin/quantity-value/unit-list
- /admin/quantity-value/convert
- /admin/quantity-value/convert-all

part of https://github.com/pimcore/studio-backend-bundle/issues/566

### CODE REVIEW

File: src/Class/Controller/QuantityValue/ConvertAllController.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    #[SuccessResponse(
+        description: 'class_quantity_value_unit_convert_all_success_response',
+        content: new JsonContent(ref: ConvertedQuantityValues::class)
+    )]
+    #[ConvertRequestBody(ConvertAllParameters::class)]
+    #[DefaultResponses([
+        HttpResponseCodes::UNAUTHORIZED,
+        HttpResponseCodes::NOT_FOUND,
+    ])]
+    public function get(#[MapRequestPayload] ConvertAllParameters $parameters): JsonResponse
```
Comment (martineiber) [inline src/Class/Controller/QuantityValue/ConvertAllController.php]: Could we add a proper action name here?

File: src/Class/Controller/QuantityValue/ConvertController.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    #[SuccessResponse(
+        description: 'class_quantity_value_unit_convert_success_response',
+        content: new ConvertedValueJson()
+    )]
+    #[ConvertRequestBody]
+    #[DefaultResponses([
+        HttpResponseCodes::UNAUTHORIZED,
+        HttpResponseCodes::NOT_FOUND,
+    ])]
+    public function get(#[MapRequestPayload] ConvertParameters $parameters): JsonResponse
```
Comment (martineiber) [inline src/Class/Controller/QuantityValue/ConvertController.php]: Same here

File: src/Class/Controller/QuantityValue/ConvertAllController.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    public function __construct(
+        SerializerInterface $serializer,
+        private readonly QuantityValueServiceInterface $quantityValueService
+
+    ) {
+        parent::__construct($serializer);
+    }
+
+    /**
+     * @throws AccessDeniedException|DatabaseException|NotFoundException|UserNotFoundException
```
Comment (martineiber) [inline src/Class/Controller/QuantityValue/ConvertAllController.php]: The `UserNotFoundException` is not getting thrown according to the `convertAllUnits()`

File: src/Class/Repository/QuantityValueRepository.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace Pimcore\Bundle\StudioBackendBundle\Class\Repository;
+
+use Pimcore\Model\DataObject\QuantityValue\Unit;
+use Pimcore\Model\DataObject\QuantityValue\Unit\Listing;
+
+/**
+ * @internal
+ */
+class QuantityValueRepository implements QuantityValueRepositoryInterface
```
Comment (martineiber) [inline src/Class/Repository/QuantityValueRepository.php]: Can be final? 
```suggestion
final readonly class QuantityValueRepository implements QuantityValueRepositoryInterface
```

**Review by martineiber — CHANGES_REQUESTED** [review]

File: src/Class/Controller/QuantityValue/ConvertAllController.php (hunk 3 above)
Comment (lukmzig) [reply src/Class/Controller/QuantityValue/ConvertAllController.php]: You are right, thank you :)

**Review by martineiber — APPROVED** [review]
LGTM :seal:

**Pull request merged**